core/src/main/java/org/jdbi/v3/core/StatementContext.java

```diff
@@ -155,6 +157,13 @@ public Object getAttribute(String key)
         return config.collectorRegistry.findCollectorFor(type);
     }
 
+    public <C extends ExtensionConfig<C>> C findConfiguration(Class<C> configType)
```

Comment:
In the case where we have an interface with multiple implementations, the `Consumer<C>` must be the most general type.  So unless your configuration interfaces' differ only in the implementation of methods, you would have no way to invoke any additional functionality on the more specific types.  e.g. if your configuration takes 
```java
interface SuperConfig { void setFoo(int); }
interface SubConfig extends SuperConfig { void setBar(int); }
```

then if you ever want to invoke `setBar` you *have* to configure `SubConfig.class` explicitly.
Do you have a good example where stacking configurations makes it simpler?  I wonder if a simple `configure(ConfigType)` and `getConfig(Class<ConfigType>)` might suffice and be easier to work with.